core/src/main/scala/org/apache/spark/api/python/PythonWorkerFactory.scala

```diff
@@ -95,10 +95,20 @@ private[spark] class PythonWorkerFactory(
   def create(): (PythonWorker, Option[Long]) = {
     if (useDaemon) {
       self.synchronized {
-        if (idleWorkers.nonEmpty) {
+        // Pull from idle workers until we one that is alive, otherwise create a new one.
+        while (idleWorkers.nonEmpty) {
           val worker = idleWorkers.dequeue()
-          worker.selectionKey.interestOps(SelectionKey.OP_READ | SelectionKey.OP_WRITE)
-          return (worker, daemonWorkers.get(worker))
+          val workerHandle = daemonWorkers(worker)
```

Comment:
Do we have a chance that this throws an exception?
If so, should we use `daemonWorkers.get(worker)` same as before, just in case?

Reply:
Since the idleWorkers queue keeps the reference alive, there should never be a case where the daemonWorker disappears until it is retrieved.